### ISSUE TITLE
Reference to System.Net.Http breaks after changing TargetFramework to net472

### ISSUE THREAD

**Issue by bottomup** [Microsoft/dotnet#737]
When I change <TargetFramework>net471</TargetFramework> to <TargetFramework>net472</TargetFramework> in my .csproj file, it can't find System.Net.Http anymore, and the build fails.

There is a nuget package reference: <PackageReference Include="System.Net.Http" Version="4.3.3" />

When in net471 mode, I can navigate to the package
// C:\Users\Bas\.nuget\packages\system.net.http\4.3.3\ref\net46\System.Net.Http.dll

When I change to net472, I can't.

All other packages are ok. The package is present as normal under Dependencies > NuGet.

I tried removing the package, clearing it from cache, reinstall it, but no luck.

**Comment by karelz** [comment]
It looks like [this known issue](https://github.com/Microsoft/dotnet/blob/master/releases/net472/KnownIssues/613745%20-%20Single-name%20references%20are%20removed%20by%20the%20SDK%20when%20targeting%204.7.2.md).
Can you please try the workaround there?

cc @joperezr

**Comment by karelz** [comment]
Are you on latest VS version?
Do you have a small repro? Does it happen for HelloWorld-style small app, or only in larger app?

@joperezr can you please help with troubleshooting? Any msbuild logs that could help?

**Comment by joperezr** [comment]
@bottomup I agree with @karelz and believe that you are hitting the issue mentioned. That said if doing the workaround suggested there doesn't work, do you mind sharing a binlog so that we can diagnose what is going on? In order to produce a binlog, simply build from the command line using `msbuild yourProject.csproj /t:rebuild /bl` and that should produce a msbuild.binlog file.

**Comment by bottomup** [comment]
Yes I am on the latest version. I don't mind sharing the binlog, but the /bl switch is unknown for my msbuild program, what can be the problem there?

**Comment by joperezr** [comment]
that probably means that the MSBuild you are trying to use is not comming from that latest msbuild. Try again on a `Developer Command Prompt for VS 2017`. If you have latest version then the flag on that MSBuild should exist and with that binlog we will be able to figure out what is the issue that you are hitting.

**Comment by bottomup** [comment]
Thanks. See attached the binlog in zip file.

[msbuild.zip](https://github.com/Microsoft/dotnet/files/1997375/msbuild.zip)

**Comment by bottomup** [comment]
Thanks. That made the difference. Any chance this workaround will be unnecessary in a future release/patch?

**Comment by joperezr** [comment]
yes, we are working on a tooling fix that will automatically add that reference for you if we detect you need it.

**Comment by joperezr** [comment]
I'll close this one since it looks like your issue is resolved and a proper fix is coming. Feel free to reopen if you feel otherwise.

**Comment by jphellemons** [comment]
I found this when I googled it with bing. But is this issue the cause for my error/issue? https://stackoverflow.com/questions/51724229/the-type-or-namespace-name-httpresponsemessage-could-not-be-found

**Comment by bottomup** [comment]
The newest NuGet version 4.3.4 release yesterday fixed the problem for me. Now waiting for Azure Service to support 4.7.2 :)
